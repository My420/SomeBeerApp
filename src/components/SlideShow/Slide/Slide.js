import React from 'react';
import PropTypes from 'prop-types';
import styles from './Slide.module.scss';

const Slide = ({ component, ...attr }) => {
  return (
    <div className={styles.slide} {...attr}>
      {component}
    </div>
  );
};

Slide.propTypes = {
  component: PropTypes.node.isRequired
};

export default Slide;
