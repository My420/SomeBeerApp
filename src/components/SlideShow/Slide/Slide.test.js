import React from 'react';
import { shallow } from 'enzyme';
import Slide from './Slide';

describe('Test <Slide /> component', () => {
  const wrapper = shallow(
    <Slide component={<span>Test</span>} data-test="test" />
  );

  test('should correct render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should  render div with  className 'slide'", () => {
    expect(wrapper.find('div').prop('className')).toBe('slide');
  });

  test('should  render component inside div', () => {
    expect(wrapper.find('div').find('span')).toHaveLength(1);
  });

  test('should transfer attribute', () => {
    expect(wrapper.find('div').prop('data-test')).toBe('test');
  });
});
