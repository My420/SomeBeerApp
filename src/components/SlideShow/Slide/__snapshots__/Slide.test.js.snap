// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Slide /> component should correct render component 1`] = `
<div
  className="slide"
  data-test="test"
>
  <span>
    Test
  </span>
</div>
`;
